hdf5-src/build.rs

```diff
@@ -0,0 +1,80 @@
+macro_rules! feature {
+    ($feature:expr) => {
+        std::env::var(concat!("CARGO_FEATURE_", $feature))
+    };
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    let mut hdf5config = cmake::Config::new("source");
+    hdf5config
+        .define("HDF5_NO_PACKAGES", "ON")
+        .define("BUILD_SHARED_LIBS", "OFF") // Only need to build a static library
+        .define("BUILD_TESTING", "OFF")
+        .define("HDF5_BUILD_TOOLS", "OFF")
+        .define("HDF5_BUILD_EXAMPLES", "OFF")
+        //
+        .define("HDF5_BUILD_JAVA", "OFF")
+        .define("HDF5_BUILD_FORTRAN", "OFF")
+        .define("HDF5_BUILD_CPP_LIB", "OFF")
+        // Default, with feature flags
+        .define("HDF5_ENABLE_PARALLEL", "OFF")
+        .define("HDF5_ENABLE_DEPRECATED_SYMBOLS", "OFF")
+        .define("HDF5_ENABLE_THREADSAFE", "OFF")
+        .define("ALLOW_UNSUPPORTED", "OFF")
+        .define("HDF5_BUILD_HL_LIB", "OFF");
+
+    if feature!("ZLIB").is_ok() {
+        let zlib_include_dir = std::env::var_os("DEP_Z_INCLUDE").unwrap();
+        let mut zlib_header = std::env::split_paths(&zlib_include_dir).nth(0).unwrap();
+        zlib_header.push("zlib.h");
+        let zlib_lib = "z";
+        hdf5config
+            .define("HDF5_ENABLE_Z_LIB_SUPPORT", "ON")
+            .define("H5_ZLIB_HEADER", &zlib_header)
+            .define("ZLIB_STATIC_LIBRARY", &zlib_lib);
+        println!("cargo:zlib_header={}", zlib_header.to_str().unwrap());
+        println!("cargo:zlib={}", zlib_lib);
+    }
+
+    if feature!("DEPRECATED").is_ok() {
+        hdf5config.define("HDF5_ENABLE_DEPRECATED_SYMBOLS", "ON");
+    }
+
+    if feature!("THREADSAFE").is_ok() {
+        hdf5config.define("HDF5_ENABLE_THREADSAFE", "ON");
+        if feature!("HL").is_ok() || feature!("PARALLEL").is_ok() {
+            println!("cargo:warning=Unsupported selection of items. This is probably ok");
+            hdf5config.define("ALLOW_UNSUPPORTED", "ON");
+        }
+    }
+
+    if feature!("HL").is_ok() {
+        #[cfg(target_os = "windows")]
+        let hdf5_hl_lib = "libhdf5_hl";
+        #[cfg(not(target_os = "windows"))]
+        let hdf5_hl_lib = "hdf5_hl";
+        println!("cargo:hl_library={}", hdf5_hl_lib);
+        hdf5config.define("HDF5_BUILD_HL_LIB", "ON");
+    }
+
+    if feature!("PARALLEL").is_ok() {
+        // TODO: Find which dependencies to link
```

Comment:
We would now require an MPI aware compiler, which we could get from `rsmpi`, or through some environment flag.

Reply:
I don't have much experience with it aside from creating bindings for mpi-enabled HDF5. It looks like in current state you won't be able to build static mpi-enabled HDF5 anyway, so there's two options - either fully work it out and test it (which is quite some work?), or drop the static parallel feature (i.e. raise cargo error if mpio and static are both enabled) for now, with a note that it can be implemented later if needed?